### DISCUSSION TITLE
open coder in a specific instance (create if not found)

### DISCUSSION THREAD

**Discussion by tuananh** [coder/coder#15592] [General]
Is this possible right now? what i want to achieve is this:

- using the same instance (ec2 for example) but populate the tools accordingly specified in devcontainer (like nixos).

**Comment by matifali** [comment]
It should be possible to check if you can check if an instance resource exists conditionally within the terraform before creating it.

[AWS template](https://github.com/coder/coder/tree/main/examples/templates/aws-linux) actually does that it creates the resource on the first launch and then in subsequent runs only changes its running state.

https://github.com/coder/coder/blob/1c08580f2358acd21f868a11d7460cc385278847/examples/templates/aws-linux/main.tf#L290-L293

Also please check #14339 and #10121
